### PR TITLE
Add a conditional implicit dependency on the import macro.

### PR DESCRIPTION
**Pull request by cfredric**
This is related to #1008 and #1013. This PR aims to improve the UX of the import macro by not requiring every target to explicitly depend on the macro.

This PR adds an implicit attribute to every `rust_library` and `rust_binary` rule, which (if the `use_real_import_macro` flag is enabled) adds the `import` macro crate to the target's `proc_macro_deps`. If the `use_real_import_macro` flag is disabled (which is the default), the implicit attribute points to a no-op shell script, and nothing is added to the target's dependencies.

### CODE REVIEW

File: util/import/BUILD.bazel [hunk 1]
```diff
@@ -1,11 +1,17 @@
 load("//rust:defs.bzl", "rust_library", "rust_proc_macro", "rust_test")
+load("//rust/private:transitions.bzl", "with_import_macro_bootstrapping_mode")
```
Comment (hlopko) [inline util/import/BUILD.bazel:2]: You can use `# buildifier: disable=bzl-visibility` to disable buildifier warning

File: rust/private/transitions.bzl [hunk 2]
```diff
@@ -18,3 +20,53 @@ wasm_bindgen_transition = transition(
...
+import_macro_dep_bootstrap_transition = transition(
+    implementation = _import_macro_dep_bootstrap_transition,
+    inputs = [],
+    outputs = ["@rules_rust//rust/settings:use_real_import_macro"],
+)
+
+def _with_import_macro_bootstrapping_mode_impl(ctx):
+    target = ctx.attr.target[0]
+    providers = [target[rust_common.crate_info], target[rust_common.dep_info]]
+    if hasattr(ctx, "executable") and hasattr(ctx.executable, "target"):
```
Comment (hlopko) [inline rust/private/transitions.bzl:52]: Could you remind me why is this necessary?

File: util/import/BUILD.bazel (hunk 1 above)
Comment (cfredric) [reply util/import/BUILD.bazel:3]: Done

File: rust/private/utils.bzl [hunk 3]
```diff
@@ -399,6 +399,21 @@ def transform_deps(deps):
         cc_info = dep[CcInfo] if CcInfo in dep else None,
     ) for dep in deps]
 
+def create_import_macro_dep(ctx):
```
[the remark this answers is not in the document]
Comment (cfredric) [reply rust/private/utils.bzl]: Done

**Review by hlopko — APPROVED** [review]
